src/sdk/object_sdk.js

```diff
@@ -245,18 +255,19 @@ class ObjectSDK {
         }
         // check for a specific bucket
         if (bucket && req.op_name !== 'put_bucket') {
-            // ANONYMOUS: cannot work without bucket, cannot work on namespace bucket (?)
+            // ANONYMOUS: cannot work without bucket.
+            // Return if the acount is anonymous
+            if (!token && process.env.NC_NSFS_NO_DB_ENV) return;
```

Comment:
Same as above - How about we add a bucketspace method that will decide if to allow anonymous requests?